packages/flutter_tools/lib/src/application_package.dart

```diff
@@ -440,8 +454,28 @@ class ApkManifestData {
     _Element launchActivity;
     for (_Element activity in activities) {
       final _Attribute enabled = activity.firstAttribute('android:enabled');
-      if (enabled == null || enabled.value.contains('0xffffffff')) {
-        launchActivity = activity;
+      final Iterable<_Element> intentFilters =
+          activity.allElements('intent-filter').cast<_Element>();
+      if (enabled!=null && !enabled.value.contains('0xffffffff'))
```

Comment:
2 things:
 - please add a blank line before the for-loop for readability
 - please extract the condition into a local variable that explains what it means.  I'm not sure what the significance is of `!enabled.value.contains('0xffffffff')`